Dev UI for GRPC fails

### Describe the bug

Navigating to Dev ui for grpc services failes with `java.util.ServiceConfigurationError: org.eclipse.microprofile.context.spi.ContextManagerExtension: io.smallrye.mutiny.context.MutinyContextManagerExtension not a subtype`

```
Listening for transport dt_socket at address: 5005
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2023-01-07 05:17:28,281 INFO  [io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread) Registering gRPC reflection service

2023-01-07 05:17:28,363 INFO  [io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread) Starting new Vert.x gRPC server ...
2023-01-07 05:17:28,489 INFO  [io.quarkus] (Quarkus Main Thread) helloworld 1.0.0-SNAPSHOT on JVM (powered by Quarkus 2.15.2.Final) started in 3.118s. Listening on: http://localhost:8080
2023-01-07 05:17:28,500 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2023-01-07 05:17:28,501 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, grpc-server, resteasy-reactive, smallrye-context-propagation, vertx]
2023-01-07 05:17:38,807 ERROR [io.qua.ver.htt.dep.dev.con.DevConsoleProcessor] ([DevConsole]vert.x-eventloop-thread-0) Dev console request failed: java.util.concurrent.CompletionException: io.quarkus.qute.TemplateException
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
        at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:674)
        at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:662)
        at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2168)
        at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:144)
        at io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.renderAsyncNoTimeout(TemplateImpl.java:218)
        at io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.renderAsync(TemplateImpl.java:191)
        at io.quarkus.vertx.http.deployment.devmode.console.DevConsole.renderTemplate(DevConsole.java:173)
        at io.quarkus.vertx.http.deployment.devmode.console.DevConsole.handle(DevConsole.java:137)
        at io.quarkus.vertx.http.deployment.devmode.console.DevConsole.handle(DevConsole.java:39)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
        at io.quarkus.vertx.http.deployment.devmode.console.FlashScopeHandler.handle(FlashScopeHandler.java:12)
        at io.quarkus.vertx.http.deployment.devmode.console.FlashScopeHandler.handle(FlashScopeHandler.java:7)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
        at io.vertx.ext.web.impl.RouterImpl.handleContext(RouterImpl.java:250)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:68)
        at io.vertx.ext.web.impl.RouterImpl.handle(RouterImpl.java:37)
        at io.quarkus.vertx.http.deployment.devmode.console.DevConsoleProcessor$2$1.handle(DevConsoleProcessor.java:195)
        at io.quarkus.vertx.http.deployment.devmode.console.DevConsoleProcessor$2$1.handle(DevConsoleProcessor.java:192)
        at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
        at io.vertx.core.impl.EventLoopContext.lambda$emit$1(EventLoopContext.java:62)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.quarkus.qute.TemplateException
        at io.quarkus.qute.CompletedStage.get(CompletedStage.java:48)
        at io.quarkus.qute.MultiResultNode.process(MultiResultNode.java:20)
        at io.quarkus.qute.MultiResultNode.process(MultiResultNode.java:20)
        at io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.lambda$renderData$5(TemplateImpl.java:233)
        at io.quarkus.qute.CompletedStage.whenComplete(CompletedStage.java:268)
        at io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.renderData(TemplateImpl.java:227)
        ... 31 more
Caused by: java.util.ServiceConfigurationError: org.eclipse.microprofile.context.spi.ContextManagerExtension: io.smallrye.mutiny.context.MutinyContextManagerExtension not a subtype
        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:593)
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1244)
        at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273)
        at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309)
        at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393)
        at java.base/java.lang.Iterable.forEach(Iterable.java:74)
        at io.smallrye.context.SmallRyeContextManager$Builder.discoverContextManagerExtensions(SmallRyeContextManager.java:318)
        at io.smallrye.context.SmallRyeContextManager$Builder.build(SmallRyeContextManager.java:335)
        at io.smallrye.context.SmallRyeContextManagerProvider.getContextManager(SmallRyeContextManagerProvider.java:48)
        at io.smallrye.context.SmallRyeContextManagerProvider.getContextManager(SmallRyeContextManagerProvider.java:37)
        at io.smallrye.context.SmallRyeContextManagerProvider.getManager(SmallRyeContextManagerProvider.java:97)
        at io.smallrye.context.SmallRyeThreadContext.getCurrentThreadContextOrDefaultContexts(SmallRyeThreadContext.java:160)
        at io.smallrye.mutiny.context.DefaultContextPropagationInterceptor.getThreadContext(DefaultContextPropagationInterceptor.java:12)
        at io.smallrye.mutiny.context.BaseContextPropagationInterceptor.decorate(BaseContextPropagationInterceptor.java:33)
        at io.smallrye.mutiny.infrastructure.Infrastructure.decorate(Infrastructure.java:153)
        at io.smallrye.mutiny.groups.MultiSubscribe.with(MultiSubscribe.java:347)
        at io.smallrye.mutiny.groups.MultiSubscribe.with(MultiSubscribe.java:321)
        at io.quarkus.grpc.runtime.health.GrpcHealthStorage.<init>(GrpcHealthStorage.java:26)
        at io.quarkus.grpc.runtime.health.GrpcHealthStorage_Bean.create(Unknown Source)
        at io.quarkus.grpc.runtime.health.GrpcHealthStorage_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
        at io.quarkus.grpc.runtime.health.GrpcHealthStorage_ClientProxy.arc$delegate(Unknown Source)
        at io.quarkus.grpc.runtime.health.GrpcHealthStorage_ClientProxy.getStatuses(Unknown Source)
        at io.quarkus.grpc.runtime.devmode.GrpcServices.entrySet(GrpcServices.java:53)
        at java.base/java.util.AbstractMap.get(AbstractMap.java:176)
        at io.quarkus.qute.ValueResolvers.mapResolveAsync(ValueResolvers.java:696)
        at io.quarkus.qute.ValueResolvers$9.resolve(ValueResolvers.java:192)
        at io.quarkus.qute.EvaluatorImpl.resolve(EvaluatorImpl.java:208)
        at io.quarkus.qute.EvaluatorImpl.resolveReference(EvaluatorImpl.java:128)
        at io.quarkus.qute.EvaluatorImpl.lambda$evaluate$0(EvaluatorImpl.java:80)
        at io.quarkus.qute.CompletedStage.thenCompose(CompletedStage.java:232)
        at io.quarkus.qute.EvaluatorImpl.evaluate(EvaluatorImpl.java:79)
        at io.quarkus.qute.ResolutionContextImpl$ChildResolutionContext.evaluate(ResolutionContextImpl.java:87)
        at io.quarkus.qute.LoopSectionHelper.resolve(LoopSectionHelper.java:45)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:58)
        at io.quarkus.qute.SectionNode$SectionResolutionContextImpl.execute(SectionNode.java:223)
        at io.quarkus.qute.InsertSectionHelper.resolve(InsertSectionHelper.java:20)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:58)
        at io.quarkus.qute.Sode$SectionResolutionContextImpl.execute(SectionNode.java:223)
        at io.quarkus.qute.SectionHelper$SectionResolutionContext.execute(SectionHelper.java:46)
        at io.quarkus.qute.Parser$1.resolve(Parser.java:1247)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)
        at io.quarkus.qute.IncludeSectionHelper.resolve(IncludeSectionHelper.java:51)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:58)
        at io.quarkus.qute.SectionNode$SectionResolutionContextImpl.execute(SectionNode.java:219)
        at io.quarkus.qute.SectionHelper$SectionResolutionContext.execute(SectionHelper.java:46)
        at io.quarkus.qute.Parser$1.resolve(Parser.java:1247)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)
        at io.quarkus.qute.SectionNode.resolve(SectionNode.java:58)
        ... 32 more
```

### Expected behavior

Dev UI for GRPC services to work correctly

### Actual behavior

Fails with Internal Server Error and  Dev console request failed

<img width="1386" alt="image" src="https://user-images.githubusercontent.com/10659853/211145676-a2a0996c-ba7c-4f83-ac02-49fab81cc126.png">


### How to Reproduce?

Steps to reproduce the behaviour:

1. create a new project `quarkus create app com.acme:helloworld`
2. Follow the grpc guide at `https://quarkus.io/guides/grpc-getting-started#implementing-a-grpc-service` adding the dependencies
```
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-grpc</artifactId>
</dependency>
```
3. Create the src/main/proto/helloworld.proto as per the guide
4. mvn compile as per the guide to generate the grpc code
5. Create the src/main/java/org/acme/HelloService.java as per the guide
6. run `quarkus dev`
7. navigate to Dev ui 
8. go to the grpc services page `http://localhost:8080/q/dev/io.quarkus.quarkus-grpc/services`

### Output of `uname -a` or `ver`

Darwin macbook 19.6.0 Darwin Kernel Version 19.6.0: Tue Jun 21 21:18:39 PDT 2022; root:xnu-6153.141.66~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17.0.3" 2022-04-19 OpenJDK Runtime Environment GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06) OpenJDK 64-Bit Server VM GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

Quarkus 2.15.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537) Maven home: /Users/priyanknarvekar/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4 Java version: 17.0.3, vendor: GraalVM Community, runtime: /Users/priyanknarvekar/.sdkman/candidates/java/22.1.0.r17-grl Default locale: en_US, platform encoding: UTF-8 OS name: "mac os x", version: "10.15.7", arch: "x86_64", family: "mac

### Additional information

_No response_

--- a/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/devmode/GrpcDevConsoleProcessor.java
+++ b/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/devmode/GrpcDevConsoleProcessor.java
@@ -56,11 +56,13 @@ import io.quarkus.gizmo.MethodCreator;
 import io.quarkus.grpc.deployment.DelegatingGrpcBeanBuildItem;
 import io.quarkus.grpc.deployment.GrpcDotNames;
 import io.quarkus.grpc.protoc.plugin.MutinyGrpcGenerator;
+import io.quarkus.grpc.runtime.config.GrpcConfiguration;
 import io.quarkus.grpc.runtime.devmode.CollectStreams;
 import io.quarkus.grpc.runtime.devmode.DelegatingGrpcBeansStorage;
 import io.quarkus.grpc.runtime.devmode.GrpcDevConsoleRecorder;
 import io.quarkus.grpc.runtime.devmode.GrpcServices;
 import io.quarkus.grpc.runtime.devmode.StreamCollectorInterceptor;
+import io.quarkus.vertx.http.runtime.HttpConfiguration;
 
 public class GrpcDevConsoleProcessor {
 
@@ -145,8 +147,9 @@ public class GrpcDevConsoleProcessor {
     @Consume(RuntimeConfigSetupCompleteBuildItem.class)
     @Record(ExecutionTime.RUNTIME_INIT)
     @BuildStep(onlyIf = IsDevelopment.class)
-    public DevConsoleRouteBuildItem createWebSocketEndpoint(GrpcDevConsoleRecorder recorder) {
-        recorder.setServerConfiguration();
+    public DevConsoleRouteBuildItem createWebSocketEndpoint(GrpcDevConsoleRecorder recorder,
+            HttpConfiguration httpConfiguration, GrpcConfiguration grpcConfiguration) {
+        recorder.setServerConfiguration(httpConfiguration, grpcConfiguration);
         return DevConsoleRouteBuildItem.builder().path("grpc-test").method("GET").handler(recorder.handler()).build();
     }
 

--- a/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/devmode/GrpcDevConsoleWebSocketListener.java
+++ b/extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/devmode/GrpcDevConsoleWebSocketListener.java
@@ -221,33 +221,40 @@ public class GrpcDevConsoleWebSocketListener implements GrpcWebSocketProxy.WebSo
         if (serviceDescriptors != null) {
             return;
         }
+        if (serverConfig == null) {
+            log.warnf("Unable to initialize client stubs for gRPC Dev UI - server config not found");
+            return;
+        }
+        if (Boolean.TRUE.equals(serverConfig.get("ssl"))) {
+            log.warnf("Unable to initialize client stubs for gRPC Dev UI - SSL is not supported");
+            return;
+        }
+
         serviceDescriptors = new HashMap<>();
         grpcClientStubs = new HashMap<>();
         try {
-            if (serverConfig == null || Boolean.FALSE.equals(serverConfig.get("ssl"))) {
-                for (Class<?> grpcServiceClass : grpcServices) {
+            for (Class<?> grpcServiceClass : grpcServices) {
 
-                    Method method = grpcServiceClass.getDeclaredMethod("getServiceDescriptor");
-                    ServiceDescriptor serviceDescriptor = (ServiceDescriptor) method.invoke(null);
-                    serviceDescriptors.put(serviceDescriptor.getName(), serviceDescriptor);
+                Method method = grpcServiceClass.getDeclaredMethod("getServiceDescriptor");
+                ServiceDescriptor serviceDescriptor = (ServiceDescriptor) method.invoke(null);
+                serviceDescriptors.put(serviceDescriptor.getName(), serviceDescriptor);
 
-                    // TODO more config options
-                    Channel channel = NettyChannelBuilder
-                            .forAddress(serverConfig.get("host").toString(), (Integer) serverConfig.get("port"))
-                            .usePlaintext()
-                            .build();
-                    Method stubFactoryMethod;
+                // TODO more config options
+                Channel channel = NettyChannelBuilder
+                        .forAddress(serverConfig.get("host").toString(), (Integer) serverConfig.get("port"))
+                        .usePlaintext()
+                        .build();
+                Method stubFactoryMethod;
 
-                    try {
-                        stubFactoryMethod = grpcServiceClass.getDeclaredMethod("newStub", Channel.class);
-                    } catch (NoSuchMethodException e) {
-                        log.warnf("Ignoring gRPC service - newStub() method not declared on %s", grpcServiceClass);
-                        continue;
-                    }
-
-                    Object stub = stubFactoryMethod.invoke(null, channel);
-                    grpcClientStubs.put(serviceDescriptor.getName(), stub);
+                try {
+                    stubFactoryMethod = grpcServiceClass.getDeclaredMethod("newStub", Channel.class);
+                } catch (NoSuchMethodException e) {
+                    log.warnf("Ignoring gRPC service - newStub() method not declared on %s", grpcServiceClass);
+                    continue;
                 }
+
+                Object stub = stubFactoryMethod.invoke(null, channel);
+                grpcClientStubs.put(serviceDescriptor.getName(), stub);
             }
         } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
             throw new IllegalStateException("Unable to initialize client stubs for gRPC Dev UI");

--- a/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/GrpcServerRecorder.java
+++ b/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/GrpcServerRecorder.java
@@ -41,6 +41,7 @@ import io.grpc.ServerInterceptors;
 import io.grpc.ServerMethodDefinition;
 import io.grpc.ServerServiceDefinition;
 import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
 import io.quarkus.arc.Subclass;
 import io.quarkus.grpc.runtime.config.GrpcConfiguration;
 import io.quarkus.grpc.runtime.config.GrpcServerConfiguration;
@@ -173,6 +174,8 @@ public class GrpcServerRecorder {
             }
         });
         shutdown.addShutdownTask(route::remove); // remove this route at shutdown, this should reset it
+
+        initHealthStorage();
     }
 
     // TODO -- handle Avro, plain text ... when supported / needed
@@ -225,12 +228,15 @@ public class GrpcServerRecorder {
     }
 
     private void initHealthStorage() {
-        GrpcHealthStorage storage = Arc.container().instance(GrpcHealthStorage.class).get();
-        storage.setStatus(GrpcHealthStorage.DEFAULT_SERVICE_NAME,
-                HealthOuterClass.HealthCheckResponse.ServingStatus.SERVING);
-        for (GrpcServiceDefinition service : services) {
-            storage.setStatus(service.definition.getServiceDescriptor().getName(),
+        InstanceHandle<GrpcHealthStorage> storageHandle = Arc.container().instance(GrpcHealthStorage.class);
+        if (storageHandle.isAvailable()) {
+            GrpcHealthStorage storage = storageHandle.get();
+            storage.setStatus(GrpcHealthStorage.DEFAULT_SERVICE_NAME,
                     HealthOuterClass.HealthCheckResponse.ServingStatus.SERVING);
+            for (GrpcServiceDefinition service : services) {
+                storage.setStatus(service.definition.getServiceDescriptor().getName(),
+                        HealthOuterClass.HealthCheckResponse.ServingStatus.SERVING);
+            }
         }
     }
 

--- a/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/devmode/GrpcDevConsoleRecorder.java
+++ b/extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/devmode/GrpcDevConsoleRecorder.java
@@ -5,13 +5,13 @@ import java.util.Map;
 
 import org.jboss.logging.Logger;
 
-import io.quarkus.arc.Arc;
-import io.quarkus.arc.InstanceHandle;
 import io.quarkus.dev.console.DevConsoleManager;
 import io.quarkus.dev.testing.GrpcWebSocketProxy;
 import io.quarkus.grpc.runtime.config.GrpcConfiguration;
 import io.quarkus.grpc.runtime.config.GrpcServerConfiguration;
 import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.vertx.http.runtime.HttpConfiguration;
+import io.quarkus.vertx.http.runtime.HttpConfiguration.InsecureRequests;
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Handler;
 import io.vertx.core.buffer.Buffer;
@@ -20,17 +20,22 @@ import io.vertx.ext.web.RoutingContext;
 
 @Recorder
 public class GrpcDevConsoleRecorder {
-    private static final Logger log = Logger.getLogger(GrpcDevConsoleRecorder.class);
 
-    public void setServerConfiguration() {
-        try (InstanceHandle<GrpcConfiguration> config = Arc.container().instance(GrpcConfiguration.class)) {
-            GrpcServerConfiguration serverConfig = config.get().server;
-            Map<String, Object> map = new HashMap<>();
+    private static final Logger LOG = Logger.getLogger(GrpcDevConsoleRecorder.class);
+
+    public void setServerConfiguration(HttpConfiguration httpConfiguration, GrpcConfiguration grpcConfiguration) {
+        GrpcServerConfiguration serverConfig = grpcConfiguration.server;
+        Map<String, Object> map = new HashMap<>();
+        if (serverConfig.useSeparateServer) {
             map.put("host", serverConfig.host);
             map.put("port", serverConfig.port);
             map.put("ssl", serverConfig.ssl.certificate.isPresent() || serverConfig.ssl.keyStore.isPresent());
-            DevConsoleManager.setGlobal("io.quarkus.grpc.serverConfig", map);
+        } else {
+            map.put("host", httpConfiguration.host);
+            map.put("port", httpConfiguration.port);
+            map.put("ssl", httpConfiguration.insecureRequests != InsecureRequests.ENABLED);
         }
+        DevConsoleManager.setGlobal("io.quarkus.grpc.serverConfig", map);
     }
 
     public Handler<RoutingContext> handler() {
@@ -39,12 +44,12 @@ public class GrpcDevConsoleRecorder {
             public void handle(RoutingContext context) {
                 context.request().toWebSocket(webSocket -> {
                     if (webSocket.failed()) {
-                        log.error("failed to connect web socket", webSocket.cause());
+                        LOG.error("failed to connect web socket", webSocket.cause());
                     } else {
                         ServerWebSocket serverWebSocket = webSocket.result();
                         Integer socketId = GrpcWebSocketProxy.addWebSocket(
                                 message -> serverWebSocket.writeTextMessage(message)
-                                        .onFailure(e -> log
+                                        .onFailure(e -> LOG
                                                 .info("failed to send back message to the gRPC Dev Console WebSocket", e)),
                                 runnable -> {
                                     if (!serverWebSocket.isClosed()) {
@@ -60,7 +65,7 @@ public class GrpcDevConsoleRecorder {
                                 });
 
                         if (socketId == null) {
-                            log.error("No gRPC dev console WebSocketListener");
+                            LOG.error("No gRPC dev console WebSocketListener");
                             serverWebSocket.close();
                             return;
                         }

